lib/libutils/ext/ftrace/ftrace.c

```diff
@@ -253,54 +255,59 @@ unsigned long __noprof ftrace_return(void)
 	else
 		return 0;
 
-	curr_buf = (char *)fbuf + fbuf->buf_off + fbuf->curr_size;
+	/* Check if we have a minimum ftrace buffer current size */
+	line_size = ENTRY_SIZE(fbuf->ret_idx);
+	if (fbuf->curr_size < line_size)
+		goto out;
+
+	line = (char *)fbuf + fbuf->buf_off + fbuf->curr_size - line_size;
 
 	/*
 	 * Check for '{' symbol as it represents if it is an exit from current
 	 * or nested function. If exit is from current function, than exit dump
 	 * via ';' symbol else exit dump via '}' symbol.
 	 */
-	if (*(curr_buf - 2) == '{') {
-		*(curr_buf - 3) = ';';
-		*(curr_buf - 2) = '\n';
-		*(curr_buf - 1) = '\0';
+	if (line[line_size - 2] == '{') {
+		line[line_size - 3] = ';';
+		line[line_size - 2] = '\n';
+		line[line_size - 1] = '\0';
 		fbuf->curr_size -= 1;
 
-		dur_loc = curr_buf - (fbuf->ret_idx +
-				      (2 * sizeof(unsigned long)) + 11);
+		dur_loc = &line[DURATION_MAX_LEN - 3];
 		ftrace_duration(dur_loc, fbuf->begin_time[fbuf->ret_idx],
 				barrier_read_counter_timer());
 	} else {
 		bool full = false;
 
-		dump_size = DURATION_MAX_LEN + fbuf->ret_idx + 3;
-		full = (fbuf->curr_size + dump_size) > fbuf->max_size;
+		line_size = EXIT_SIZE(fbuf->ret_idx);
+		full = (fbuf->curr_size + line_size) > fbuf->max_size;
 		if (full)
-			full = !fbuf_make_room(fbuf, dump_size);
+			full = !fbuf_make_room(fbuf, line_size);
 
 		if (!full) {
-			curr_buf = (char *)fbuf + fbuf->buf_off +
-				   fbuf->curr_size;
+			line = (char *)fbuf + fbuf->buf_off + fbuf->curr_size;
 
-			for (i = 0; i < (DURATION_MAX_LEN + fbuf->ret_idx); i++)
+			for (i = 0; i < (DURATION_MAX_LEN + fbuf->ret_idx);
+			     i++) {
```

Comment:
can drop inner parentheses. 
nice thing about it is that this fits in a single line.

Reply:
Done.